src/image/image.js

```diff
@@ -1712,6 +1712,8 @@ export class Image {
             new Index(label.centroidIndex));
           label.volume = label.count * pixelVolume;
           label.unit = volumeUnit;
+          // diameters should be already in the correct units
+          label.diameterUnit = lengthUnit;
```

Comment:
the labels object starts to be quite complex, could you add a class definition, at least for what the image class sends out in the 'labelschanged' event?
I'm also wondering if always having 'values' (for ex diameter) as {value, unit} would make things clearer.

Reply:
Added type definition, and updated values to contain units.